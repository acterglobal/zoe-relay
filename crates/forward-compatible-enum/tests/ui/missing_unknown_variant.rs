use forward_compatible_enum::ForwardCompatibleEnum;

#[derive(ForwardCompatibleEnum)]
pub enum MissingUnknownVariant {
    #[discriminant(1)]
    Variant1,
    #[discriminant(2)]
    Variant2,
    // Missing Unknown variant!
}