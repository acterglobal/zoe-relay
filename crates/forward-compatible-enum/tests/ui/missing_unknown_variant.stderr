error: Enum must contain an Unknown variant named 'Unknown'
  --> tests/ui/missing_unknown_variant.rs:4:1
   |
4  | / pub enum MissingUnknownVariant {
5  | |     #[discriminant(1)]
6  | |     Variant1,
7  | |     #[discriminant(2)]
...  |
10 | | }
   | |_^

error[E0601]: `main` function not found in crate `$CRATE`
  --> tests/ui/missing_unknown_variant.rs:10:2
   |
10 | }
   |  ^ consider adding a `main` function to `$DIR/tests/ui/missing_unknown_variant.rs`
